From edf52212a09bd80b52dc9932b5ca19e20dfcaa2b Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Sat, 18 Oct 2014 10:52:49 -0700
Subject: [PATCH 1/2] Fix some clang integer sign/size mismatch warnings

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 src/cursor.c  | 4 ++--
 src/library.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/cursor.c b/src/cursor.c
index 92bd417..23c3283 100644
--- a/src/cursor.c
+++ b/src/cursor.c
@@ -226,7 +226,7 @@ static Bool
 _XcursorHeckbertMedianCut (const XcursorImage *image, XcursorCoreCursor *core)
 {
     XImage	    *src_image = core->src_image, *msk_image = core->msk_image;
-    int		    npixels = image->width * image->height;
+    unsigned int    npixels = image->width * image->height;
     int		    ncolors;
     int		    n;
     XcursorPixel    *po, *pn, *pc;
@@ -395,7 +395,7 @@ _XcursorFloydSteinberg (const XcursorImage *image, XcursorCoreCursor *core)
     int		    *aPicture, *iPicture, *aP, *iP;
     XcursorPixel    *pixel, p;
     int		    aR, iR, aA, iA;
-    int		    npixels = image->width * image->height;
+    unsigned int    npixels = image->width * image->height;
     int		    n;
     int		    right = 1;
     int		    belowLeft = image->width - 1;
diff --git a/src/library.c b/src/library.c
index 48b75bf..f08e8f0 100644
--- a/src/library.c
+++ b/src/library.c
@@ -49,7 +49,7 @@ XcursorLibraryPath (void)
 static  void
 _XcursorAddPathElt (char *path, const char *elt, int len)
 {
-    int	    pathlen = strlen (path);
+    size_t    pathlen = strlen (path);
 
     /* append / if the path doesn't currently have one */
     if (path[0] == '\0' || path[pathlen - 1] != '/')
-- 
2.8.3

